apps/ejabberd/src/jid.erl

```diff
@@ -42,25 +42,22 @@
 -spec make(User :: ejabberd:user(), Server :: ejabberd:server(),
            Resource :: ejabberd:resource()) ->
     ejabberd:jid()  | error.
+
 make(User, Server, Resource) ->
-    case nodeprep(User) of
-        error -> error;
-        LUser ->
-            case nameprep(Server) of
-                error -> error;
-                LServer ->
-                    case resourceprep(Resource) of
-                        error -> error;
-                        LResource ->
-                            #jid{user = User,
-                                 server = Server,
-                                 resource = Resource,
-                                 luser = LUser,
-                                 lserver = LServer,
-                                 lresource = LResource}
-                    end
-            end
-    end.
+  case {nodeprep(User), nameprep(Server), resourceprep(Resource)} of
```

Comment:
The previous version is a bit more efficient when it comes to parsing wrong jids. Previously nameprep was run only if nodeprep passed. And similarly resourceprep. Error cases doesn't happen that offten so maybe may note can be ignored.

Reply:
I think that unless someone sends malformed JIDs on purpose, it is ver unlikely in normal operation to get errors here.